prediction-applier/src/test/java/org/alfresco/hxi_connector/prediction_applier/hxinsight/PredictionListenerTest.java

```diff
@@ -0,0 +1,121 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.prediction_applier.hxinsight;
+
+import static org.apache.camel.builder.AdviceWith.adviceWith;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+
+import static org.alfresco.hxi_connector.prediction_applier.hxinsight.PredictionListener.ROUTE_ID;
+
+import java.util.stream.Stream;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.apache.camel.CamelContext;
+import org.apache.camel.FluentProducerTemplate;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.model.ToDefinition;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import org.alfresco.hxi_connector.common.model.prediction.Prediction;
+import org.alfresco.hxi_connector.common.model.repository.Node;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class PredictionListenerTest
+{
+    private static final String TEST_ENDPOINT = "direct:predictions-queue";
+    private static final String MOCK_ENDPOINT = "mock:direct-nodes-endpoint";
+
+    CamelContext camelContext;
```

Comment:
Any particular reason this field and 3 following ones are package-private (could be private I guess)?

Reply:
This is something that @tpage-alfresco taught me - from the test perspective components access level is not important and this, at least in theory should make the test a bit easier to analyze and understand

Reply:
In the other hand I must admit, that at the beginning this way of writing seemed strange to me, but now I'm already used to it

Reply:
I'm still not consistent in this. In theory this makes sense to me, but it seems harder to remember _not_ to write private!

Reply:
In the other hand it's a good habit write the `private` keyword